services/py-api/src/service/participants_service.py

```diff
@@ -35,10 +35,29 @@ async def _create_participant_and_team_in_transaction(
             return participant
 
         return Ok((participant.ok_value, team.ok_value))
-
+    
     async def register_admin_participant(
         self, input_data: ParticipantRequestBody
-    ) -> Ok[Tuple[Participant, Team]] | Err[DuplicateEmailError | DuplicateTeamNameError | Exception]:
-        # TODO: Add Capacity check 2
+    ) -> Result[Tuple[Participant, Team], DuplicateEmailError | DuplicateTeamNameError| HackathonCapacityExceededError | Exception]:
+        # Capacity Check 2
+        has_capacity = await self.check_capacity_register_admin_participant_case()
+        if not has_capacity:
+            return Err(HackathonCapacityExceededError(self._team_repo.MAX_NUMBER_OF_TEAMS_IN_HACKATHON))
+
+        # Proceed with registration if there is capacity
         result = await self._tx_manager.with_transaction(self._create_participant_and_team_in_transaction, input_data)
         return result
+
+    async def check_capacity_register_admin_participant_case(self) -> bool:
```

Comment:
`_check_capacity_register_admin_participant_case` and move it before `register_admin_participant`

Reply:
Ivan, in your comment just below, you ask to fix the tests and write unit tests for "this" function, as well as, "And this case in the handler". 

Can you elaborate on which tests should I fix, and what does "And this case in the handler" means. Also, do I understand correctly that I need to write a unit test for "_check_capacity_register_admin_participant_case"?

Reply:
@Alexandr-Ignatov, again, as I said in the group in Messenger, there was a problem with your commits, talk with @Isak-Bego about how to proceed with this PR. I suggest we just close it, delete this branch, and start all over again, as there aren't a lot of changes, and it has become a mess. But @Isak-Bego will tell.

Reply:
Fixed ✅✅✅✅